client/cli/src/commands/runcmd.rs

```diff
@@ -562,13 +563,34 @@ fn interface_str(
 	}
 }
 
-/// Default to verbosity level 0, if none is provided.
+/// Utility error for returning custom errors that implement the std::error::Error trait.
+#[derive(Debug)]
+struct TelemetryParsingError {
+    details: String
+}
+
+impl TelemetryParsingError {
+    fn new(msg: &str) -> TelemetryParsingError {
+        TelemetryParsingError{details: msg.to_string()}
+    }
+}
+
+impl fmt::Display for TelemetryParsingError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl Error for TelemetryParsingError {
+    fn description(&self) -> &str {
```

Comment:
I don't think that's necessary any more. They only relevant thing in `Error` trait is `fn source` I believe, the rest should be left as-is.

Reply:
Cool, thanks for the tips! :D